Revit_Core_Engine/Create/FamilyInstance/FamilyInstance.cs

```diff
@@ -155,6 +157,104 @@ private static FamilyInstance FamilyInstance_OneLevelBased(Document document, Fa
+            }
+
+            if (host == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Element could not be created. Host element not found.");
+                return null;
+            }
+
+            LocationCurve location = host.Location as LocationCurve;
+            Line line = location.Curve as Line;
+            if (line == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Element could not be created. Host element type does not support line breaking functionality.");
+                return null;
+            }
+            
+            if (line.Distance(origin) > settings.DistanceTolerance)
+            {
+                if (origin == null)
+                {
+                    BH.Engine.Reflection.Compute.RecordError("Element could not be created. Its location point is could not be projected on the host MEPCurve");
+                    return null;
+                }
+                origin = line.Project(origin).XYZPoint;
+                BH.Engine.Reflection.Compute.RecordWarning("Origin point location does not lie on the host MEPCurve. Closest point has been projected.");
+            }
+            
+            //BreakCurve
+            ElementId newElementId;
+            if (host is Pipe)
+            {
+                newElementId = PlumbingUtils.BreakCurve(document, host.Id, origin);
+            }
+            else if (host is Duct)
+            {
+                newElementId = MechanicalUtils.BreakCurve(document, host.Id, origin);
+            }
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("Element could not be created. Host element type does not support line breaking functionality.");
+                return null;
+            }
+
+            //Find connectors
+            Element newElement = document.GetElement(newElementId);
+            MEPCurve newElementMEPCurve = newElement as MEPCurve;
+            ConnectorSet hostConnSet = host.ConnectorManager.Connectors;
+            ConnectorSet newElConnSet = newElementMEPCurve.ConnectorManager.Connectors;
+            Connector conn1 = null;
+            Connector conn2 = null;
+            foreach (Connector hostConn in hostConnSet)
+            {
+                foreach (Connector newElConn in newElConnSet)
+                {
+                    if (hostConn.Origin.IsAlmostEqualTo(newElConn.Origin))
+                    {
+                        conn1 = hostConn;
+                        conn2 = newElConn;
+                        break;
+                    }
+                }
+            }
+
+            //Place family instance
+            FamilyInstance familyInstance = document.Create.NewUnionFitting(conn1, conn2);
+            if (!familyInstance.SetParameter(BuiltInParameter.ELEM_TYPE_PARAM, familySymbol.Id))
+                BH.Engine.Reflection.Compute.RecordWarning("Family Symbol type is not correct for Host element.");
```

Comment:
I would opt for something more generic, like:
```suggestion
                BH.Engine.Reflection.Compute.RecordWarning("The input family type could not be assigned to the created instance, possibly because it is not applicable to the host element.");
```